### DISCUSSION TITLE
Why parser ignores single quotes?

### DISCUSSION THREAD

**Discussion by eugenelepekhin** [antlr/antlr4#4049] [Q&A] [answered]
Hi,
I have the following lexer file FusionLexer.g4:
```
lexer grammar FusionLexer;
NumberLiteral: DecNumber;
fragment DecNumber: NonzeroDigit DecDigit*;
fragment DecDigit: [0-9];
fragment NonzeroDigit: [1-9];
Whitespace: [ \t\r\n]+ -> skip;

```
And the parser file FusionParser.g4:
```
parser grammar FusionParser;
fusionProgram: number* EOF;
number: NumberLiteral;

```
When I parse the following text:
```
'8 '9
'10

```
I am getting a valid parse tree:
`(fusionProgram (number 8) (number 9) (number 10) <EOF>)
`
Which is puzzling me as I'd expect a syntax error as quotes are not allowed anywhere in the grammar.
Can someone shed some light on what is going on here?
Thanks,
Eugene

**Comment by kaby76** [comment, marked as answer]
The parser syntax error count ([here](https://github.com/antlr/antlr4/blob/76fa05c21b12b96a6c12a0a82e611ed9d87d5af4/runtime/Java/src/org/antlr/v4/runtime/Parser.java#L154)) isn't incremented for lexer errors. The lexer is a different recognizer than the parser, but doesn't seem to keep a similar error counter, which is rather confusing. The way I deal with this is to replace the error listeners for both the parser and lexer with my own error listeners, which have completely independent error counters that I can use.

**Reply by eugenelepekhin** [reply]
Awesome, thank you.
